src/Git.hs

```diff
@@ -146,6 +147,11 @@ data TagResult
 data FetchWithTags = WithTags | NoTags
   deriving stock (Eq, Show)
 
+data ReasonToFail 
+  = WrongFixups 
+  | OtherReason 
```

Comment:
The other reason's would be `MergeFailed` and `RebaseFailed`, I think it'd be nicer to make that explicit, especially since we're basing a user comment based on the specific failure.